optimum/intel/utils/import_utils.py

```diff
@@ -196,8 +170,108 @@ def is_openvino_available():
+            None,
+            (
+                "pip install",
+                "--pre" if is_dev_version else "",
+                "-U",
+                f"{package_name}=={compatible_version}",
+                "--extra-index-url https://storage.openvinotoolkit.org/simple/wheels/nightly"
+                if is_dev_version
+                else "",
+            ),
+        )
+    )
+
+
+@functools.lru_cache(1)
 def is_openvino_tokenizers_available():
-    return _openvino_tokenizers_available
+    if not is_openvino_available():
+        return False
+
+    try:
+        import openvino_tokenizers
+    except ImportError:
+        ov_major_version = _openvino_version.split("-")[0]
+        logger.info(
+            "OpenVINO Tokenizers is not available, tokenizer won't be converted. To automatically "
+            "convert tokenizer during OpenVINO model export install OpenVINO Tokenizers.\n"
+            "For PyPI distribution of OpenVINO:\n"
+            "pip install openvino-tokenizers[transformers]\n"
+            "For pre-release OpenVINO version:\n"
+            "pip install --pre -U openvino openvino-tokenizers"
+            "--extra-index-url https://storage.openvinotoolkit.org/simple/wheels/nightly\n"
+            "If you using archive OpenVINO distribution install release, nightly version or build from source:\n"
+            f"Release: pip install --no-deps openvino-tokenizers~={ov_major_version}\n"
+            f"Nightly: pip install --pre openvino-tokenizers~={ov_major_version}.0.dev --extra-index-url "
+            f"https://storage.openvinotoolkit.org/simple/wheels/nightly\n"
+            "Build from source instructions: https://github.com/openvinotoolkit/openvino_tokenizers"
+            "?tab=readme-ov-file#build-and-install-from-source\n"
+        )
+        return False
+
+    try:
+        pip_metadata_version = importlib_metadata.version("openvino")
+        is_nightly = False
+    except importlib_metadata.PackageNotFoundError:
+        try:
+            pip_metadata_version = importlib_metadata.version("openvino-nightly")
+            is_nightly = True
+        except importlib_metadata.PackageNotFoundError:
+            pip_metadata_version = False
+            is_nightly = False
+
+    try:
+        openvino_tokenizers._get_factory()
+    except RuntimeError as e:
+        tokenizers_version = openvino_tokenizers.__version__
+
+        if tokenizers_version == "0.0.0.0":
+            try:
+                tokenizers_version = importlib_metadata.version("openvino_tokenizers") or tokenizers_version
+            except importlib_metadata.PackageNotFoundError:
+                pass
+        message = (
+            "OpenVINO and OpenVINO Tokenizers versions are not binary compatible.\n"
+            f"OpenVINO version:            {_openvino_version}\n"
+            f"OpenVINO Tokenizers version: {tokenizers_version}\n"
+            "First 3 numbers should be the same. Update OpenVINO Tokenizers to compatible version. "

```

Comment:
![image](https://github.com/huggingface/optimum-intel/assets/77325899/b2132d91-8b66-4691-87e8-fd61e6b6783b)
Since the error message said "first 3 numbers" I expected these slightly different nightly versions would work, but it doesn't. So then it makes more sense to say that the versions have to be exactly the same.

I get this error even when I use an archive of March 27 https://storage.openvinotoolkit.org/repositories/openvino/packages/nightly/2024.1.0-14876-fc5da9d9359 and openvino tokenizers  2024.1.0.0.dev20240327 
![image](https://github.com/huggingface/optimum-intel/assets/77325899/641e3faf-6045-41c9-9a7b-1051601558c2)

Reply:
This is because of the different Linux distros. Add this message if Linux platform is detected:
> The PyPI version of OpenVINO Tokenizers is built on CentOS and may not be compatible with other Linux distributions; rebuild OpenVINO Tokenizers from source.